make-xges.py

```diff
@@ -27,11 +27,57 @@ def file_to_uri(path):
     path = os.path.realpath(path)
     return 'file://' + path
 
+def my_time_format(raw_time):
+
+    # total amount of seconds
+    seconds = 0.0
+
+    if raw_time is None:
+
+        seconds = None
+
+    else:
+
+        # allow an empty value
+        if raw_time == '':
+            raw_time = '0'
+
+        # seconds should be always 0
+        # minutes should be always 1 ecc.
+        parts = raw_time.split(':')
+        parts.reverse()
+        n = len(parts)
+
+        # seconds (ss) - also float is fine here
+        if( n > 0 ):
+            seconds = float( parts[0] )
+
+        # minutes (mm:ss)
+        if( n > 1 ):
+            seconds += float( parts[1] ) * 60.0
+
+        # hours (hh:mm:ss)
+        if( n > 2 ):
+            seconds += float( parts[2] ) * 3600.0
+
+        # days (dd:hh:mm:ss)
+        if( n > 3 ):
+            seconds += float( parts[3] ) * 86400.0
```

Comment:
We probably only expect integers for these components, right? Unless it make sense to accept values like `1.5:30`?

Reply:
Yup. But if the result is a float I usually prefer float components. What do you suggest instead?